core-plugins/src/main/java/co/cask/hydrator/plugin/common/KinesisUtil.java

```diff
@@ -0,0 +1,136 @@
+
+import com.amazonaws.AmazonServiceException;
+import com.amazonaws.services.kinesis.AmazonKinesisClient;
+import com.amazonaws.services.kinesis.model.CreateStreamRequest;
+import com.amazonaws.services.kinesis.model.DescribeStreamRequest;
+import com.amazonaws.services.kinesis.model.ResourceNotFoundException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Utility methods to help create and manage Kinesis Streams
+ */
+public class KinesisUtil {
+
+  private static final Logger LOG = LoggerFactory.getLogger(KinesisUtil.class);
+
+  /**
+   * Creates an Amazon Kinesis stream if it does not exist and waits for it to become available
+   *
+   * @param kinesisClient The {@link AmazonKinesisClient} with Amazon Kinesis read and write privileges
+   * @param streamName The Amazon Kinesis stream name to create
+   * @param shardCount The shard count to create the stream with
+   * @throws IllegalStateException Invalid Amazon Kinesis stream state
+   * @throws IllegalStateException Stream does not go active before the timeout
+   */
+  public static void createAndWaitForStreamToBecomeAvailable(AmazonKinesisClient kinesisClient,
+                                                             String streamName,
+                                                             int shardCount) {
+    if (streamExists(kinesisClient, streamName)) {
+      String state = streamState(kinesisClient, streamName);
+      switch (state) {
+        case "DELETING":
+          long waitTime = System.currentTimeMillis() + 1000 * 120;
+          while (System.currentTimeMillis() < waitTime && streamExists(kinesisClient, streamName)) {
+            try {
+              LOG.info("...Deleting Stream {} ...", streamName);
+              Thread.sleep(1000 * 10);
+            } catch (InterruptedException e) {
+            }
+          }
+          if (streamExists(kinesisClient, streamName)) {
+            LOG.error("KinesisUtils timed out waiting for stream {} to delete", streamName);
+            throw new IllegalStateException(String.format("KinesisUtils timed out waiting for stream {} to delete",
+                                                          streamName));
+          }
+          break;
+        case "ACTIVE":
+          LOG.info("Stream {} is ACTIVE", streamName);
+          return;
+        case "CREATING":
+          break;
+        case "UPDATING":
+          LOG.info("Stream {} is UPDATING", streamName);
+          return;
+        default:
+          throw new IllegalStateException("Illegal stream state: " + state);
+      }
+    } else {
+      CreateStreamRequest createStreamRequest = new CreateStreamRequest();
+      createStreamRequest.setStreamName(streamName);
+      createStreamRequest.setShardCount(shardCount);
+      kinesisClient.createStream(createStreamRequest);
+      LOG.info("Stream {} created", streamName);
+    }
+    long waitTime = System.currentTimeMillis() + (10 * 60 * 1000);
```

Comment:
Creation takes more time than deletion, how ever the SLEEP_INTERVAL is to loop on sleeps to wait. It would have been nice if kinesis would have provided out of the box waitTillCreate method